src/main/kotlin/biokotlin/util/MergeGVCFUtil.kt

```diff
@@ -0,0 +1,361 @@
+            .map { allele ->
+                when (allele) {
+                    "." -> Allele.NO_CALL
+
+                    "REF" -> refAllele
+
+                    else -> alleleMap[allele] ?: throw IllegalArgumentException("Allele not found: $allele")
+                }
+            }
+
+        GenotypeBuilder(info.samples[index], alleleObjs)
+            .phased(phased)
+            .make()
+
+    }
+
+    return VariantContextBuilder()
+        .source(".")
+        .alleles(alleleMap.values)
+        .chr(info.position.contig)
+        .start(info.position.position.toLong())
+        .stop(info.position.position.toLong())
+        .genotypes(genotypes)
+        .make()
+
+}
+
+var timeMinPosition = 0L
+var timeNextLowestPosition = 0L
+var timeLowestLookAheadStart = 0L
+var timeResult = 0L
+var timeAdvanceVariant = 0L
+private fun nextPosition(gvcfReaders: Array<VCFReader>, currentPosition: Position? = null): Position? {
+
+    val minPosition: Position
+    val sortedVariants: List<VCFReader>
+    measureNanoTime {
+        // Sort current variants by start position
+        // Get the minimum start position
+        sortedVariants = gvcfReaders.filter { it.variant() != null }.sortedBy { it.variant()!!.positionRange }
+        if (sortedVariants.isEmpty()) return null
+        minPosition = sortedVariants.first().variant()!!.startPosition
+    }.let { timeMinPosition += it }
+
+    // If no current position, return the minimum position
+    if (currentPosition == null) return minPosition
+
+    val nextLowestPosition: Position?
+    measureNanoTime {
+        nextLowestPosition = sortedVariants
+            .find { it.variant()!!.startPosition > currentPosition }?.variant()?.startPosition
+    }.let { timeNextLowestPosition += it }
+
+    val lowestLookAheadStart: Position?
+    measureNanoTime {
+        lowestLookAheadStart = sortedVariants
+            .filter { it.lookAhead() != null }
+            .minByOrNull { it.lookAhead()!!.startPosition }
+            ?.lookAhead()?.startPosition
+    }.let { timeLowestLookAheadStart += it }
+
+    val result: Position?
+    measureNanoTime {
+        result = when {
+            nextLowestPosition == null -> lowestLookAheadStart
+            lowestLookAheadStart == null -> nextLowestPosition
+            else -> minOf(nextLowestPosition, lowestLookAheadStart)
+        }
+    }.let { timeResult += it }
+
+    result?.let {
+        measureNanoTime {
+            sortedVariants.filter { it.variant()!!.endPosition < result }.forEach {
+                it.advanceVariant()
+            }
+        }.let { timeAdvanceVariant += it }
+        return result
+    } ?: return null
+
+}
+
+/**
+ * Creates a HTSJDK reference allele.
+ */
+private fun alleleRef(allele: String): Allele {
+    return Allele.create(allele, true)
+}
+
+/**
+ * Creates a HTSJDK alternate allele.
+ */
+private fun alleleAlt(allele: String): Allele {
+    return Allele.create(allele, false)
+}
+
+fun main() {
```

Comment:
Probably should remove this main()

Reply:
Yes, that was removed. Yes unit tests to come. Still sorting out the details and checking correct output against GATK